### ISSUE TITLE
Consider pointing binder icon to latest release branch

### ISSUE THREAD

**Issue by jarmarshall** [DiODeProject/MuMoT#320]
Currently the binder icon in `README.md` and `getting_started.rst` point to the master branch. It may be better to have these point to the most recent release branch instead. If this is agreed then the following actions need to be taken:

- [ ] add a list of previous binder references, possibly under `mybinder.org` in `about.rst`
- [ ] update `development.rst` to include instructions on bumping the binder branch reference in `README.md` and `getting_started.rst`, and adding the previous binder branch reference to the list of previous releases, to enable ongoing access

**Comment by willfurnass** [comment]
Yes, could have separate mybinder buttons/links for the latest release and for the HEAD of `master`.

**Comment by willfurnass** [comment]
mybinder.org links in README and docs will now start a Jupyter session for MuMoT v1.1.1 instead of `HEAD` of `master` (PR #369).  

Still need to address the points above.
